src/ReverseProxy/Service/Config/ProxyRouteTransformExtensions.cs

```diff
@@ -7,108 +11,107 @@ namespace Microsoft.ReverseProxy.Service
     public static class ProxyRouteTransformExtensions
     {
         /// <summary>
-        /// Adds a transform to the route which set the request path with the given value.
+        /// Clones the ProxyRoute and adds the transform.
         /// </summary>
-        public static void AddTransformPathSet(this ProxyRoute proxyRoute, PathString path)
+        /// <returns>The cloned route with the new transform.</returns>
+        public static ProxyRoute WithTransform(this ProxyRoute proxyRoute, IReadOnlyDictionary<string, string> transform)
         {
-            proxyRoute.Transforms ??= new List<IDictionary<string, string>>();
+            var transforms = proxyRoute.Transforms == null ?
+                new List<IReadOnlyDictionary<string, string>>()
+                : new List<IReadOnlyDictionary<string, string>>(proxyRoute.Transforms);
```

Comment:
```suggestion
            List<IReadOnlyDictionary<string, string>> transforms;
            if (proxyRoute.Transforms == null)
            {
                transforms = new List<IReadOnlyDictionary<string, string>>();
            }
            else
            {
                transforms = new List<IReadOnlyDictionary<string, string>>(proxyRoute.Transforms.Count + 1);
                transforms.AddRange(proxyRoute.Transforms);
            }
```
Does this suggestion worth the effort to prevent the possible Resizing the underlying array of the list?